lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -68,4 +99,97 @@ def desired_options
       AWSResource.lookup_options(options, resource: new_resource)
     end
   end
+
+  def update_attachment(volume)
+    status = volume.status
+    #
+    # If we were told to attach the volume to a machine, do so
+    #
+    if desired_instance.is_a?(AWS::EC2::Instance)
+      case status
+      when :in_use
+        attachment = volume.attachments.first
+        # wrong instance
```

Comment:
This comment seems valid - whether the volume is attached to the wrong instance, or has the wrong device name (/dev/sda1 vs /dev/sda2) we need to update the attachment.  This means calling detach and then attach, correct?